app/src/main/java/org/schabi/newpipe/util/external_communication/TextLinkifier.java

```diff
@@ -244,14 +284,33 @@ private static void changeIntentsOfDescriptionLinks(final TextView textView,
 
             for (final URLSpan span : urls) {
                 final String url = span.getURL();
-                final ClickableSpan clickableSpan = new ClickableSpan() {
-                    public void onClick(@NonNull final View view) {
-                        if (!InternalUrlsHandler.handleUrlDescriptionTimestamp(
-                                new CompositeDisposable(), context, url)) {
-                            ShareUtils.openUrlInBrowser(context, url, false);
+                final ClickableSpan clickableSpan;
+                if (addLongPressListenerToLinks) {
+                    clickableSpan = new LongPressClickableSpan() {
+                        @Override
+                        public void onClick(@NonNull final View view) {
+                            if (!InternalUrlsHandler.handleUrlDescriptionTimestamp(
+                                    new CompositeDisposable(), context, url)) {
```

Comment:
Also in other places: creating a `new CompositeDisposable()` defeats the purpose of adding things to a disposable, since disposables should be cleared somewhere when the app/activity/fragment/dialog closes
```suggestion
                                    disposables, context, url)) {
```